src/skin/qml/qmllibraryproxy.cpp

```diff
@@ -0,0 +1,25 @@
+#include "skin/qml/qmllibraryproxy.h"
+
+#include <QAbstractItemModel>
+
+#include "library/library.h"
+#include "skin/qml/qmllibrarytracklistmodel.h"
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+QmlLibraryProxy::QmlLibraryProxy(Library* pLibrary, QObject* parent)
+        : QObject(parent),
+          m_pLibrary(pLibrary),
+          m_pModel(
+                  new QmlLibraryTrackListModel(m_pLibrary->trackTableModel())) {
+}
+
+QmlLibraryProxy::~QmlLibraryProxy() {
+    delete m_pModel;
+}
```

Comment:
parented_ptr?

Reply:
I don't think that works with member properties.

Reply:
not sure, I just thought it would suit the usecase of parented_ptr but I don't really understand how it works or how to use it. I guess @daschuer should know.

Reply:
I just tried it and it works: https://github.com/mixxxdj/mixxx/commit/eff29a67989b95e45b0c5de5faaa8fb1dfe4a6f3